src/vs/workbench/contrib/terminal/browser/terminal.ts

```diff
@@ -80,6 +81,28 @@ export interface IQuickPickTerminalObject {
 	keyMods: IKeyMods | undefined
 }
 
+export interface TerminalCommand {
+	command: string;
+	timestamp: number;
+	getOutput: () => string | undefined;
+	cwd?: string;
+	exitCode?: number;
+	marker?: IMarker;
+}
+
+export interface ICommandTracker {
+	scrollToPreviousCommand(): void;
+	scrollToNextCommand(): void;
+	selectToPreviousCommand(): void;
+	selectToNextCommand(): void;
+	selectToPreviousLine(): void;
+	selectToNextLine(): void;
+	get commands(): TerminalCommand[];
+	get cwds(): string[];
```

Comment:
Prefer readonly properties to getters in the interface, the consumer doesn't care how the property is implemented

```suggestion
	readonly commands: TerminalCommand[];
	readonly cwds: string[];
	scrollToPreviousCommand(): void;
	scrollToNextCommand(): void;
	selectToPreviousCommand(): void;
	selectToNextCommand(): void;
	selectToPreviousLine(): void;
	selectToNextLine(): void;
```